docs/source/upgrade.md

```diff
@@ -2,6 +2,31 @@
 
 This pages describes Scylla Operator upgrade procedures.
 
+## `v1.1.0` -> `v1.2.0`
+
+1.2.0 release brought a lot of changes to the Scylla Operator deployment process.
+To properly update Scylla Operator one must delete old objects and install updated ones.
+
+A rolling restart of your Scylla clusters is expected during the upgrade procedure.
+
+1. Checkout source code of v1.2.0:
+   ```
+   git checkout v1.2.0
+   ```
+1. Remove old scylla operator namespace - in our case it's called `scylla-operator-system`:
+    ```
+    kubectl delete namespace scylla-operator-system --cascade=foreground
```

Comment:
why cascade foreground?

Reply:
to ensure that non-leader-election aware statefulset from 1.1 is deleted before new deployment is installed.

Reply:
`kubectl delete` waits for the object (namespace) being removed which implies waiting for `kubernetes` finalizer on the namespace which ensures GC has deleted all objects in the namespace. I guess to ensure someone doesn't have a prehistoric kubectl we could add explicit `--wait=true` to fail if the feature is not present.

Doesn't cascade deletion affect only children and the objects in the namespace don't have an owner reference to the namespace itself?